### PR TITLE
Remove internal attachments on ninjs formatter

### PR DESCRIPTION
**Pull request by pablopunk**
SDFID-574

This is going to branch `release/1.32`

### CODE REVIEW

File: superdesk/publish/formatters/ninjs_formatter.py [hunk 1]
```diff
@@ -426,16 +427,17 @@ def _format_attachments(self, article):
...
+            if is_attachment_public(attachment): # don't save internal attachments
+              output.append({
+                  'id': str(attachment['_id']),
+                  'title': attachment['title'],
+                  'description': attachment['description'],
+                  'filename': attachment['filename'],
+                  'mimetype': attachment['mimetype'],
+                  'length': attachment.get('length'),
+                  'media': str(attachment['media']),
+                  'href': '/assets/{}'.format(str(attachment['media'])),
```
Comment (petrjasek) [inline superdesk/publish/formatters/ninjs_formatter.py]: it's better to use `app.media.url_for_media`

File: superdesk/metadata/item.py [hunk 2]
```diff
@@ -646,6 +646,15 @@
     'CropBottom': {'type': 'integer'}
 }
 
+def is_attachment_public(attachment):
```
Comment (petrjasek) [inline superdesk/metadata/item.py]: imo this should be in `superdesk.attachments` module

File: superdesk/publish/formatters/ninjs_formatter.py (hunk 1 above)
Comment (pablopunk) [reply superdesk/publish/formatters/ninjs_formatter.py]: you mean instead `href`? cause I didn't change any of this code, I just indented it under the `if`

File: superdesk/metadata/item.py (hunk 2 above)
Comment (pablopunk) [reply superdesk/metadata/item.py]: agree

File: superdesk/metadata/item.py [hunk 3]
```diff
@@ -654,14 +663,8 @@ def remove_metadata_for_publish(item):
...
 
-    def filter_internal_attachment(attachment):
-        if attachment.get('attachment'): # retrieve object reference
-            attachment = superdesk.get_resource_service('attachments').find_one(req=None, _id=attachment['attachment'])
-        return not attachment.get('internal')
-
-    # Remove attachments that are internal
     if len(item.get('attachments', [])) > 0:
-        item['attachments'] = filter(filter_internal_attachment, item['attachments'])
+        item['attachments'] = list(filter(is_attachment_public, item['attachments']))
```
Comment (jerome-poisson) [inline superdesk/metadata/item.py]: the pythonic way is to use list comprehension here:
```python
[attachment for attachment in item['attachments'] if is_attachment_public(attachment)]
```

File: superdesk/publish/formatters/ninjs_formatter.py (hunk 1 above)
Comment (pablopunk) [reply superdesk/publish/formatters/ninjs_formatter.py]: should I change it anyway then?

Comment (petrjasek) [reply superdesk/publish/formatters/ninjs_formatter.py]: yes pls

Comment (pablopunk) [reply superdesk/publish/formatters/ninjs_formatter.py]: is this ok?

```
'href': app.media.url_for_media(attachment['media']),
```

**Pull request merged**